java/src/main/java/com/plaid/quickstart/resources/TransactionsResource.java

```diff
@@ -1,60 +1,47 @@
 package com.plaid.quickstart.resources;
 
 import java.io.IOException;
-import java.util.Date;
+import java.time.LocalDate;
 
-import com.fasterxml.jackson.annotation.JsonProperty;
-
-import com.plaid.client.PlaidClient;
-import com.plaid.client.request.TransactionsGetRequest;
-import com.plaid.client.response.ErrorResponse;
-import com.plaid.client.response.ItemStatus;
-import com.plaid.client.response.TransactionsGetResponse;
+import com.plaid.client.request.PlaidApi;
+import com.plaid.client.model.TransactionsGetRequest;
+import com.plaid.client.model.TransactionsGetRequestOptions;
+import com.plaid.client.model.TransactionsGetResponse;
 import com.plaid.quickstart.QuickstartApplication;
 
 import javax.ws.rs.GET;
-import javax.ws.rs.POST;
 import javax.ws.rs.Path;
 import javax.ws.rs.Produces;
 import javax.ws.rs.core.MediaType;
 
-import jersey.repackaged.com.google.common.base.Throwables;
 import retrofit2.Response;
 
 @Path("/transactions")
 @Produces(MediaType.APPLICATION_JSON)
 public class TransactionsResource {
-  private final PlaidClient plaidClient;
+  private final PlaidApi plaidClient;
+
 
-  public TransactionsResource(PlaidClient plaidClient) {
+  public TransactionsResource(PlaidApi plaidClient) {
     this.plaidClient = plaidClient;
   }
 
   @GET
   public TransactionsGetResponse getTransactions() throws IOException {
-    String accessToken = QuickstartApplication.accessToken;
-    Date startDate = new Date(System.currentTimeMillis() - 86400000L * 100);
-    Date endDate = new Date();
-
-    TransactionsGetRequest request =
-      new TransactionsGetRequest(accessToken, startDate, endDate)
-        .withCount(100);
-
-    Response<TransactionsGetResponse> response = null;
-    for (int i = 0; i < 5; i++) {
-      response = plaidClient.service().transactionsGet(request).execute();
-      if (response.isSuccessful()) {
-        break;
-      } else {
-        try {
-          ErrorResponse errorResponse = plaidClient.parseError(response);
-          Thread.sleep(3000);
-        } catch (InterruptedException e) {
-          throw Throwables.propagate(e);
-        }
-      }
-    }
-
-    return response.body();
+    LocalDate startDate = LocalDate.now().minusDays(365 * 2);
```

Comment:
So initially there was a `retry` here, and it looks like there isn't in the new revision.
Does this fail sometime? Is the retry necessary?

Reply:
I don't THINK the retry is necessary, but I added it back in.  (I didn't do it before cause it involved error handling, and I was gonna come back to that...)